### DISCUSSION TITLE
Usage in Functions / Triggers

### DISCUSSION THREAD

**Discussion by marvindinges** [supabase/supabase#31284] [Questions]
I'm using this https://github.com/supabase-community/supabase-custom-claims to implement custom claims for the self-hosted instance.

I have this function / trigger to set a few values in my public.users table. I also want to set a role here using the set_claim
```sql
CREATE OR REPLACE FUNCTION public.handle_new_auth_user()
RETURNS TRIGGER AS $$
BEGIN
  INSERT INTO public.users (auth_id, name)
  VALUES (NEW.id, NEW.raw_user_meta_data ->> 'user_name');
  
  SELECT set_claim(NEW.id, 'role', '"USER"');
  
  RETURN NEW;
END;
$$ LANGUAGE plpgsql SECURITY DEFINER;

CREATE OR REPLACE TRIGGER after_user_created
AFTER INSERT ON auth.users
FOR EACH ROW
EXECUTE FUNCTION public.handle_new_auth_user();
``` 

But it doesn't work. What am I doing wrong here?

**Comment by GaryAustin1** [comment]
What does not work mean?  Did you look for errors in the Postgres log?
If set_claim() is in the public schema you likely need to do public.set_claim(xxx).  There should be an error though saying something about that.
